src/bin/pg_checksums/pg_checksums.c

```diff
@@ -513,16 +529,25 @@ main(int argc, char *argv[])
 				break;
 			case 'D':
 				DataDir = optarg;
+				strcpy(FrontendHookPgDataPath, optarg);
```

Comment:
`strncpy` is safer here.
[MAX_ARG_STRLEN](https://stackoverflow.com/a/4185165)

> And as additional limit, one argument must not be longer than MAX_ARG_STRLEN (131072).

same thing applies to the below `strcpy`

Reply:
MAX_ARG_STRLEN is in `<linux/binfmts.h>` and to get `PAGE_SIZE` we need -D_DEFAULT_SOURCE.

I am switching to strlcpy